src/main/java/org/elasticsearch/rest/action/support/RestTable.java

```diff
@@ -96,11 +96,35 @@ public static RestResponse buildTextPlainResponse(Table table, RestChannel chann
         return new BytesRestResponse(RestStatus.OK, BytesRestResponse.TEXT_CONTENT_TYPE, bytesOut.bytes());
     }
 
-    private static List<DisplayHeader> buildDisplayHeaders(Table table, RestRequest request) {
+    static List<DisplayHeader> buildDisplayHeaders(Table table, RestRequest request) {
         String pHeaders = request.param("h");
         List<DisplayHeader> display = new ArrayList<>();
         if (pHeaders != null) {
-            for (String possibility : Strings.splitStringByCommaToArray(pHeaders)) {
+            Map<String, Table.Cell> headerMap = table.getHeaderMap();
+            Collection<String> headers = new LinkedHashSet<>(table.getHeaders().size());
+            // check headers and aliases
+            for (String header : Strings.splitStringByCommaToArray(pHeaders)) {
+                if (Regex.isSimpleMatchPattern(header)) {
+                    for (Map.Entry<String, Table.Cell> configuredHeaderEntry : headerMap.entrySet()) {
+                        String configuredHeader = configuredHeaderEntry.getKey();
+                        if (Regex.simpleMatch(header, configuredHeader)) {
+                            headers.add(configuredHeader);
+                        } else if (configuredHeaderEntry.getValue().attr.containsKey("alias")) {
+                            String[] aliases = Strings.splitStringByCommaToArray(configuredHeaderEntry.getValue().attr.get("alias"));
+                            for (String alias : aliases) {
+                                if (Regex.simpleMatch(header, alias)) {
+                                    headers.add(configuredHeader);
+                                    break;
+                                }
+                            }
+                        }
+                    }
+                } else {
+                    headers.add(header);
+                }
```

Comment:
wondering if extracting this block to a new method would make it a bit more readable, I have no strong opinion though